drivers/gpio/gpio_stm32.c

```diff
@@ -163,6 +163,10 @@ int gpio_stm32_configure(u32_t *base_addr, int pin, int conf, int altf)
 		}
 	}
 
+#if defined(CONFIG_SOC_SERIES_STM32L4X) && defined(GPIO_ASCR_ASC0)
```

Comment:
A comment in the code explaining why we're doing this would be nice.

Reply:
Can you move the comment inside the #if def ?
I prefer this way because it could happen sometime that people add code between comment and code so we don't know anymore what is the comment related to. If comment is inside the ifdef this is less likely to happen.

Also, please rephrase to "For STM32L47xx/48xx, register ASCR should be configured  to connect analog switch of gpio lines to the ADC."